utils.py

```diff
@@ -119,40 +129,138 @@ def generate_signal(status, chandle, func='PICO_SINE', **kwargs):
+                                                                           len(cur_conds), 
+                                                                           cur_mode
+            )
+            assert_pico_ok(status['setTrigConds'])
+
+            # extract the trigger properties and trigger directions that are set later            
+            cur_dirs = [cur_trig[1] for cur_trig in cur_conj]
+            cur_props = [cur_trig[2] for cur_trig in cur_conj]
+
+            trigger_props += cur_props
+            trigger_dirs += cur_dirs
+        else:
+            break
 
-def set_trigger(status, chandle, source, trigger_thrs_mV, resolution, range_V, direction = 'RISING_OR_FALLING', dummy = False):
+        ind += 1
+
+    pico_trigger_dirs = (structs.PICO_DIRECTION * len(trigger_dirs))(*trigger_dirs)
+    status['setTrigDir'] = ps.ps6000aSetTriggerChannelDirections(handle,
+                                                                 ctypes.byref(pico_trigger_dirs),
+                                                                 len(trigger_dirs)
+    )
+    assert_pico_ok(status['setTrigDir'])
+
+    auxOutputEnable = 0 # AUX trigger output is not supported :-(
+    pico_trigger_props = (structs.PICO_TRIGGER_CHANNEL_PROPERTIES * len(trigger_props))(*trigger_props)
+    status['setTrigProps'] = ps.ps6000aSetTriggerChannelProperties(handle, 
+                                                                   ctypes.byref(pico_trigger_props), 
+                                                                   len(trigger_props), 
+                                                                   auxOutputEnable, 
+                                                                   autoTriggerMicroSeconds
+    )
+    assert_pico_ok(status['setTrigProps'])
+
+def set_simple_trigger(status, handle, source, trigger_thrs_mV, resolution, channel_range, direction = 'RISING_OR_FALLING', dummy = False):
     '''
     Method to setup a trigger
     '''
-
-    channel_range = channel_ranges[f'PICO_{range_V}']
     
     # get max ADC value
     min_ADC = ctypes.c_int16()
     max_ADC = ctypes.c_int16()
     status['getAdcLimits'] = ps.ps6000aGetAdcLimits(
-        chandle,
+        handle,
         resolution,
         ctypes.byref(min_ADC),
         ctypes.byref(max_ADC)
     )
     assert_pico_ok(status['getAdcLimits'])
 
     # set simple trigger
-    pico_direction = enums.PICO_THRESHOLD_DIRECTION[f'PICO_{direction}']
+    pico_direction = enums.PICO_THRESHOLD_DIRECTION[direction]
+    pico_channel_range = PICO_CONNECT_PROBE_RANGE[channel_range]
+
     status['setSimpleTrigger'] = ps.ps6000aSetSimpleTrigger(
-        chandle,
+        handle,
         0 if dummy else 1,
         source,
-        mV2adc(trigger_thrs_mV, channel_range, max_ADC),
+        mV2adc(trigger_thrs_mV, pico_channel_range, max_ADC),
         pico_direction,
         0,  # delay = 0 s
-        1000000  # autoTriggerMicroSeconds
+        0  # autoTriggerMicroSeconds
```

Comment:
is it ok to put 0 microseconds here? Is it needed to reduce the dead time?

Reply:
Aha! When I tested the advanced triggering functionality, I found that setting an autotrigger timeout of zero makes it wait indefinitely (this is consistent with the documentation). I then tried whether `setSimpleTrigger` results in the same behaviour, and it does. (Nothing is stated in the documentation, except that `setSimpleTrigger` calls internally the functions that set up the "advanced" trigger---so that seems consistent.)

Long story short, I turned it into a configurable parameter now, which should be better in any case.

Reply:
Thanks for the explanation! 👍